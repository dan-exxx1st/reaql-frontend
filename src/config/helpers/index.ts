import { useState, useEffect } from 'react';

export function useLoaded({ src }: { src?: HTMLImageElement['src'] }) {
	const [loaded, setLoaded] = useState('false');

	useEffect(() => {
		if (!src) {
			return undefined;
		}

		setLoaded('loading');

		let active = true;
		const image = new Image();
		image.src = src;
		image.onload = () => {
			if (!active) {
				return;
			}
			setLoaded('loaded');
		};
		image.onerror = () => {
			if (!active) {
				return;
			}
			setLoaded('error');
		};

		return () => {
			active = false;
		};
	}, [src]);

	return loaded;
}
