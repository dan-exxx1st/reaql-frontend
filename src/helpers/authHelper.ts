import { Session, User } from 'lib/graphql/types';
import { useContext } from 'react';
import { UserContext } from './contexts/userContext';

export const CheckAuth = () => {
    const { state } = useContext(UserContext);
    const sessionString = localStorage.getItem('session');
    if (sessionString || state?.user?.email) {
        return true;
    }
    return false;
};

export const getAuthTokens = () => {
    const session = localStorage.getItem('session');
    if (session) {
        const {
            accessToken,
            refreshToken,
        }: { accessToken: string; refreshToken: string } = JSON.parse(session);

        return { accessToken, refreshToken };
    } else {
        return { accessToken: '', refreshToken: '' };
    }
};

export const setSession = (session: {
    refreshToken: string;
    accessToken: string;
}) => {
    const sessionData = {
        accessToken: session.accessToken,
        refreshToken: session.refreshToken,
    };
    const sessionString = JSON.stringify(sessionData);
    localStorage.setItem('session', sessionString);
};

export const setUserAndSession = (payload: {
    user: User;
    session?: Session | null;
}) => {
    const { user, session } = payload;
    if (session) setSession(session);
    localStorage.setItem('userEmail', user.email);
};

export const clearSession = () => {
    localStorage.clear();
    window.location.reload();
};
