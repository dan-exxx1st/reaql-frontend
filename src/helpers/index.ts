import { parse } from 'querystring';

export function getDialogIdFromSearch(search: string) {
    const params = parse(search);

    const dialogId = Object.entries(params)
        .map(([key, value]) => key.indexOf('dialog') !== -1 && value)
        .toString();

    return dialogId;
}
