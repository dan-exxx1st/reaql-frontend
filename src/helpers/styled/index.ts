import colorTheme from './colors';
import fontTheme from './fonts';

export { colorTheme, fontTheme };

const effects = {
    formBoxShadow: '5px 5px 15px rgba(37, 37, 37, 0.1)',
};

export default {
    ...colorTheme,
    ...fontTheme,
    ...effects,
};
