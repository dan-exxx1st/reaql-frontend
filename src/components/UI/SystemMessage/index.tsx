import React, { FC } from 'react';
import { StyledSystemMessageText, StyledSystemMessageWrapper } from './style';

import { ISystemMessageProps } from 'lib/types/components/UI';

const SystemMessage: FC<ISystemMessageProps> = (props) => {
    const { children } = props;
    return (
        <StyledSystemMessageWrapper {...props}>
            <StyledSystemMessageText variant="body2">
                {children}
            </StyledSystemMessageText>
        </StyledSystemMessageWrapper>
    );
};

export default SystemMessage;
