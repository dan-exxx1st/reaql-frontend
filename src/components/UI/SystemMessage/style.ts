import styled from 'styled-components';

import { Typography, Wrapper } from 'components/UI';
import { ISystemMessageProps } from 'lib/types/components/UI';

export const StyledSystemMessageWrapper = styled(Wrapper)<ISystemMessageProps>`
    padding: 15px;
    justify-content: center;

    box-shadow: 0px 3px 7px 6px rgba(0, 0, 0, 0.1);
    border-radius: 5px;
    width: 250px;

    ${({ variant }) => {
        let color = '';
        if (variant === 'success') color = '#4caf50';
        if (variant === 'error') color = '#f44336';
        if (variant === 'warning') color = '#ff9800';
        if (variant === 'info') color = '#2196f3';

        return `background: ${color}`;
    }};
`;

export const StyledSystemMessageText = styled(Typography)`
    color: ${({ theme }) => theme.wc};
    font-weight: 500;
    letter-spacing: 1.1px;
`;
