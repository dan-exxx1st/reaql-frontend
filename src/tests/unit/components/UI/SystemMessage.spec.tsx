import React from 'react';
import SystemMessage from 'components/UI/SystemMessage';
import { renderWithTheme } from 'tests/helpers/withTheme';

describe('<SystemMessage />', () => {
    it('Should render a success message correctly', () => {
        const {
            container: { firstChild },
        } = renderWithTheme(
            <SystemMessage variant="success" className="test-class">
                Test msg
            </SystemMessage>
        );
        expect(firstChild).toMatchSnapshot();
    });

    it('Should render an error message correctly', () => {
        const {
            container: { firstChild },
        } = renderWithTheme(
            <SystemMessage variant="error">Test msg</SystemMessage>
        );
        expect(firstChild).toMatchSnapshot();
    });

    it('Should render a warning message correctly', () => {
        const {
            container: { firstChild },
        } = renderWithTheme(
            <SystemMessage variant="warning">Test msg</SystemMessage>
        );
        expect(firstChild).toMatchSnapshot();
    });

    it('Should render an info message correctly', () => {
        const {
            container: { firstChild },
        } = renderWithTheme(
            <SystemMessage variant="info">Test msg</SystemMessage>
        );
        expect(firstChild).toMatchSnapshot();
    });
});
