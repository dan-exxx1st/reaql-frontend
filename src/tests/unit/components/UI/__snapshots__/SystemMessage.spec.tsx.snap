// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SystemMessage /> Should render a success message correctly 1`] = `
.c2 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  padding: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0px 3px 7px 6px rgba(0,0,0,0.1);
  border-radius: 5px;
  width: 250px;
  background: #4caf50;
}

.c3 {
  color: #FAFAFA;
  font-weight: 500;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
}

<div
  class="c0 c1 test-class"
>
  <p
    class="c2 c3"
  >
    Test msg
  </p>
</div>
`;

exports[`<SystemMessage /> Should render a warning message correctly 1`] = `
.c2 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  padding: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0px 3px 7px 6px rgba(0,0,0,0.1);
  border-radius: 5px;
  width: 250px;
  background: #ff9800;
}

.c3 {
  color: #FAFAFA;
  font-weight: 500;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
}

<div
  class="c0 c1"
>
  <p
    class="c2 c3"
  >
    Test msg
  </p>
</div>
`;

exports[`<SystemMessage /> Should render an error message correctly 1`] = `
.c2 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  padding: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0px 3px 7px 6px rgba(0,0,0,0.1);
  border-radius: 5px;
  width: 250px;
  background: #f44336;
}

.c3 {
  color: #FAFAFA;
  font-weight: 500;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
}

<div
  class="c0 c1"
>
  <p
    class="c2 c3"
  >
    Test msg
  </p>
</div>
`;

exports[`<SystemMessage /> Should render an info message correctly 1`] = `
.c2 {
  font-family: 'Roboto',sans-serif;
  font-weight: normal;
  font-size: 14px;
  -webkit-letter-spacing: 0.25px;
  -moz-letter-spacing: 0.25px;
  -ms-letter-spacing: 0.25px;
  letter-spacing: 0.25px;
  color: #000000;
  margin: 0;
}

.c0 {
  padding: 0;
  margin: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  width: auto;
  height: auto;
}

.c1 {
  padding: 15px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  box-shadow: 0px 3px 7px 6px rgba(0,0,0,0.1);
  border-radius: 5px;
  width: 250px;
  background: #2196f3;
}

.c3 {
  color: #FAFAFA;
  font-weight: 500;
  -webkit-letter-spacing: 1.1px;
  -moz-letter-spacing: 1.1px;
  -ms-letter-spacing: 1.1px;
  letter-spacing: 1.1px;
}

<div
  class="c0 c1"
>
  <p
    class="c2 c3"
  >
    Test msg
  </p>
</div>
`;
